src/CommandLineUtils/Utilities/Prompt.cs

```diff
@@ -89,36 +92,105 @@ public static string GetString(string prompt, string defaultValue = null, Consol
+            {
+                switch (key)
+                {
+                    case Backspace:
+                        resp.Remove(resp.Length - 1, 1);
+                        break;
+                    default:
+                        resp.Append(key);
+                        break;
+                }
+            }
+
+            return resp.ToString();
+        }
+
+#if !NETSTANDARD1_6
+        /// <summary>
+        /// Gets a response as a SecureString object. Input is masked with an asterisk.
+        /// </summary>
+        /// <param name="prompt">The question to display on the command line</param>
+        /// <param name="promptColor">The console color to use for the prompt</param>
+        /// <param name="promptBgColor">The console background color for the prompt</param>
+        /// <returns>A finalized SecureString object, may be empty.</returns>
+        public static System.Security.SecureString GetPasswordAsSecureString(string prompt, ConsoleColor? promptColor = null, ConsoleColor? promptBgColor = null)
+        {
+            var secureString = new System.Security.SecureString();
+
+            foreach (var key in ReadObfuscatedLine(prompt, promptColor, promptBgColor))
+            {
+                switch (key)
+                {
+                    case Backspace:
+                        secureString.RemoveAt(secureString.Length - 1);
+                        break;
+                    default:
+                        secureString.AppendChar(key);
+                        break;
+                }
+            }
+
+            secureString.MakeReadOnly();
+            return secureString;
+        }
+#endif
+
+        /// <summary>
+        /// Base implementation of GetPassword and GetPasswordAsString. Prompts the user for
+        /// a password and yields each key as the user inputs. Password is masked as input. Pressing Escape will reset the password
+        /// by flushing the stream with backspace keys.
+        /// </summary>
+        /// <param name="prompt">The question to display on the command line</param>
+        /// <param name="promptColor">The console color to use for the prompt</param>
+        /// <param name="promptBgColor">The console background color for the prompt</param>
+        /// <returns>A stream of characters as input by the user including Backspace for deletions.</returns>
+        private static IEnumerable<char> ReadObfuscatedLine(string prompt, ConsoleColor? promptColor = null, ConsoleColor? promptBgColor = null)
+        {
+            const string whiteOut = "\b \b";
             Write(prompt, promptColor, promptBgColor);
             Console.Write(' ');
 
-            var resp = new StringBuilder();
+            var readChars = 0;
             ConsoleKeyInfo key;
-
             do
             {
                 key = Console.ReadKey(intercept: true);
+                // TODO: What should happen if the key has a modifier?
```

Comment:
We should handle this. If the Control or Alt modifier is set, we should ignore the key. I can do that in a follow-up change since my original code didn't do this correctly.